contracts/libraries/Validator.sol

```diff
@@ -44,27 +44,36 @@ library Validator {
     {
         require(sigs.length == 130, "two transcation signatures, 65 bytes each, are required");
         require(confirmSignatures.length % 65 == 0, "confirm signatures must be 65 bytes in length");
-        
+
         bytes memory sig0 = slice(sigs, 0, 65);
         if (input1) {
             require(confirmSignatures.length == 130, "two confirm signatures required with two inputs");
             bytes memory sig1 = slice(sigs, 65, 65);
 
             // check both input signatures
-            return recover(txHash, sig0) == recover(confirmationHash, slice(confirmSignatures, 0, 65)) &&
-                recover(txHash, sig1) == recover(confirmationHash, slice(confirmSignatures, 65, 65));
+            address recoveredTx0 = recover(txHash, sig0);
+            address recoveredConfirmation0 = recover(confirmationHash, slice(confirmSignatures, 0, 65));
+
+            address recoveredTx1 = recover(txHash, sig1);
+            address recoveredConfirmation1 = recover(confirmationHash, slice(confirmSignatures, 65, 65));
+
+            return recoveredTx0 == recoveredConfirmation0 && recoveredTx1 == recoveredConfirmation1 &&
+                ((recoveredTx0 != address(0) && recoveredConfirmation0 != address(0)) ||
+                (recoveredTx1 != address(0) && recoveredConfirmation1 != address(0)));
```

Comment:
There are redundant checks in the above. If recoveredTx0 == recoveredConfirmation0 then recoveredTx0 != address(0) is a sufficient check (recoveredConfirmation0 != address(0) is unnecessary).

Reply:
Addressed